### PR TITLE
Test app for posts and comments.

### PR DESCRIPTION
**Pull request by ahsanarbisoft**
For posts list view the URL will be "http://localhost:18000/mysite/" And for admin go to Home › Mysite.

### CODE REVIEW

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2533/workflows/workflows_id/jobs/2533/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/mysite/__init__.py (100%)
openedx/adg/lms/mysite/admin.py (100%)
openedx/adg/lms/mysite/apps.py (100%)
openedx/adg/lms/mysite/constants.py (100%)
openedx/adg/lms/mysite/models.py (90.5%): Missing lines 24,38
openedx/adg/lms/mysite/tests/factories.py (100%)
openedx/adg/lms/mysite/tests/test_models.py (100%)
openedx/adg/lms/mysite/tests/test_views.py (100%)
openedx/adg/lms/mysite/urls.py (100%)
openedx/adg/lms/mysite/views.py (100%)
-------------
Total:   93 lines
Missing: 2 lines
Coverage: 97%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2550/workflows/workflows_id/jobs/2550/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/mysite/__init__.py (100%)
openedx/adg/lms/mysite/admin.py (100%)
openedx/adg/lms/mysite/apps.py (100%)
openedx/adg/lms/mysite/constants.py (100%)
openedx/adg/lms/mysite/models.py (90.0%): Missing lines 31,44
openedx/adg/lms/mysite/tests/factories.py (100%)
openedx/adg/lms/mysite/tests/test_models.py (100%)
openedx/adg/lms/mysite/tests/test_views.py (100%)
openedx/adg/lms/mysite/urls.py (100%)
openedx/adg/lms/mysite/views.py (93.2%): Missing lines 33-34,71
-------------
Total:   130 lines
Missing: 5 lines
Coverage: 96%
-------------
```

File: openedx/adg/lms/mysite/migrations/0002_auto_20210812_1050.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Generated by Django 2.2.17 on 2021-08-12 10:50
+
+import django.core.validators
+from django.db import migrations, models
+import django.utils.timezone
+import model_utils.fields
+import openedx.adg.lms.mysite.helpers
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/mysite/migrations/0002_auto_20210812_1050.py:7]: There is no helper file in PR. can you please verify that?

Comment (javeriaejaz-arbisoft) [reply openedx/adg/lms/mysite/migrations/0002_auto_20210812_1050.py:7]: Also, we shouldn't be creating 3 migration files. Revert these migrations and create a single migration file.

File: openedx/adg/lms/mysite/models.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    Model for posts
+    """
+
+    title = models.CharField(max_length=MAX_POST_TITLE_LENGTH, unique=True, verbose_name=_('Post title'))
+    description = models.TextField(null=True, blank=True, verbose_name=_('Description'))
+    image = models.ImageField(
+        upload_to='team-members/images/', verbose_name=_('Image'),
+        validators=[FileExtensionValidator(ALLOWED_PROFILE_IMAGE_EXTENSIONS)],
+        help_text=_('You can add only png or jpg or jpeg or svg images'),
+        blank=True
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/mysite/models.py:23]: Can we please set image field as required? Also, if we were to add image field as optional, then we should set null=True along with blank=True.

File: openedx/adg/lms/mysite/models.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        return f'{self.title}'
+
+
+class Comment(TimeStampedModel):
+    """
+    Model for post's comments
+    """
+
+    post = models.ForeignKey(Post, on_delete=models.CASCADE)
+    comment = models.CharField(max_length=MAX_COMMENT_TEXT_LENGTH, verbose_name=_('Comment'))
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/mysite/models.py:40]: `text` would be a better field name here. Similar model and field name can become confusing in further queries.

File: openedx/adg/lms/mysite/views.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    """
+
+    template_name = 'adg/lms/mysite/list.html'
+    model = Post
+    context_object_name = 'posts'
+    ordering = ['-created']
+    paginate_by = 5
+
+    # pylint: disable=W0221
+    def get_context_data(self, **kwargs):
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/mysite/views.py:24]: Please add docstring here.

File: openedx/adg/lms/mysite/views.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        context = super(PostListView, self).get_context_data(**kwargs)
+        posts = self.get_queryset()
+        page = self.request.GET.get('page')
+        paginator = Paginator(posts, self.paginate_by)
+        try:
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/mysite/views.py:29]: Why do we need try/except here?

File: openedx/adg/lms/mysite/views.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+class PostUpdateView(UpdateView):
+    """
+    View to render the posts update page
+    """
+
+    template_name = 'adg/lms/mysite/update.html'
+    model = Post
+    context_object_name = 'post'
+    fields = ('title', 'description', )
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/mysite/views.py:68]: Why have we not added image here?

File: openedx/adg/lms/mysite/admin.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+
+@admin.register(Comment)
+class CommentAdmin(admin.ModelAdmin):
+    """
+    Django admin class for Comment
+    """
+
+    list_display = ('id', 'comment', 'votes', 'created', 'modified')
+    list_filter = ['created']
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/mysite/admin.py:33]: We shouldn't be using list_filter on created field.

File: openedx/adg/lms/mysite/constants.py [hunk 8]
```diff
@@ -0,0 +1,6 @@
+"""
+Constants for mysite app
+"""
+MAX_POST_TITLE_LENGTH = 100
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/mysite/constants.py:4]: Can we increase it at least 150?

File: openedx/adg/lms/mysite/models.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+from .constants import ALLOWED_PROFILE_IMAGE_EXTENSIONS, MAX_COMMENT_TEXT_LENGTH, MAX_POST_TITLE_LENGTH
+
+
+class Post(TimeStampedModel):
+    """
+    Model for posts
+    """
+
+    title = models.CharField(max_length=MAX_POST_TITLE_LENGTH, unique=True, verbose_name=_('Post title'))
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/mysite/models.py:17]: verbose name can only be `Title`.

File: openedx/adg/lms/mysite/models.py [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    description = models.TextField(null=True, blank=True, verbose_name=_('Description'))
+    image = models.ImageField(
+        upload_to='team-members/images/', verbose_name=_('Image'),
+        validators=[FileExtensionValidator(ALLOWED_PROFILE_IMAGE_EXTENSIONS)],
+        help_text=_('You can add only png or jpg or jpeg or svg images'),
+        blank=True
+    )
+
+    class Meta:
+        verbose_name = _('Posts Comment')
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/mysite/models.py:27]: For post model, verbose name `Post` seems more appropriate rather than `Post Comment`

**Review by javeriaejaz-arbisoft — CHANGES_REQUESTED** [review]

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2565/workflows/workflows_id/jobs/2565/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/mysite/__init__.py (100%)
openedx/adg/lms/mysite/admin.py (100%)
openedx/adg/lms/mysite/apps.py (100%)
openedx/adg/lms/mysite/constants.py (100%)
openedx/adg/lms/mysite/models.py (90.0%): Missing lines 32,45
openedx/adg/lms/mysite/tests/factories.py (100%)
openedx/adg/lms/mysite/tests/test_models.py (100%)
openedx/adg/lms/mysite/tests/test_views.py (100%)
openedx/adg/lms/mysite/urls.py (100%)
openedx/adg/lms/mysite/views.py (92.0%): Missing lines 36-37,74,97
-------------
Total:   135 lines
Missing: 6 lines
Coverage: 95%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2566/workflows/workflows_id/jobs/2566/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/mysite/__init__.py (100%)
openedx/adg/lms/mysite/admin.py (100%)
openedx/adg/lms/mysite/apps.py (100%)
openedx/adg/lms/mysite/constants.py (100%)
openedx/adg/lms/mysite/models.py (90.0%): Missing lines 32,45
openedx/adg/lms/mysite/tests/factories.py (100%)
openedx/adg/lms/mysite/tests/test_models.py (100%)
openedx/adg/lms/mysite/tests/test_views.py (100%)
openedx/adg/lms/mysite/urls.py (100%)
openedx/adg/lms/mysite/views.py (92.0%): Missing lines 36-37,74,97
-------------
Total:   135 lines
Missing: 6 lines
Coverage: 95%
-------------
```

**Comment by asadali145** [comment]
Closing the PR as we are done with its purpose of learning.

**Pull request closed without merging**